server/src/main/java/io/druid/client/selector/HighestPriorityTierSelectorStrategy.java

```diff
@@ -47,6 +39,6 @@ public HighestPriorityTierSelectorStrategy(@JacksonInject ServerSelectorStrategy
   @Override
   public Comparator<Integer> getComparator()
   {
-    return comparator;
+    return REVESE_COMPARATOR;
```

Comment:
It's `Comparator.reverseOrder()`

Reply:
I couldn't get `Integer::compare.reverseOrder()` or similar to work. Every kind of syntax fixing ended up being much dirtier looking compared to the way it is here. I'm open to suggestions though.

Reply:
probably natural order, let me try that one

Reply:
It's a polymorphic static method, https://docs.oracle.com/javase/8/docs/api/java/util/Comparator.html#reverseOrder--, you should be able to write just `Comparator.reverseOrder()`.

Reply:
fixed